packages/server/graphql/mutations/helpers/getUsersToIgnore.ts

```diff
@@ -1,24 +1,42 @@
-import {DataLoaderWorker} from '../../graphql'
-import getRethink from '../../../database/rethinkDriver'
+import getRedis from '../../../utils/getRedis'
+import {UserPresence} from '../../intranetSchema/mutations/connectSocket'
 
 // finds all the users who are at the same url
-const getUsersToIgnore = async (viewerId: string, teamId: string, dataLoader: DataLoaderWorker) => {
-  const r = await getRethink()
-  const viewer = await dataLoader.get('users').load(viewerId)
-  const lastSeenAtURL = viewer.lastSeenAtURL || ''
-  if (!lastSeenAtURL.startsWith('/meet')) return []
-  return (r
-    .table('TeamMember')
-    .getAll(teamId, {index: 'teamId'})
-    .filter({isNotRemoved: true})
-    .merge((teamMember) => ({
-      lastSeenAtURL: r
-        .table('User')
-        .get(teamMember('userId'))('lastSeenAtURL')
-        .default('~')
-    }))
-    .filter({lastSeenAtURL})('userId')
-    .run() as unknown) as string[]
+const getUsersToIgnore = async (viewerId: string, teamId: string) => {
+  const redis = getRedis()
+  let userPresence
+  let activeTeamMemberIds
+  await redis
+    .multi()
+    .lrange(`presence:${viewerId}`, 0, -1)
+    .smembers(`team:${teamId}`)
+    .exec((execErr, results) => {
+      if (execErr) throw new Error(`Failed to execute redis command: ${execErr}`)
+      results.forEach((res, index) => {
+        if (index === 0 && !res[0]) {
```

Comment:
This looks a bit ugly to me but `results` are an array of `[err, response]` so I think these checks are necessary. I haven't found any cleaner examples of reading from `exec` after many a Google.